src/hotspot/share/runtime/threads.cpp

```diff
@@ -1417,9 +1419,29 @@ JavaThread *Threads::owning_thread_from_monitor_owner(ThreadsList * t_list,
   return the_owner;
 }
 
+JavaThread* Threads::owning_thread_from_object(ThreadsList * t_list, oop obj) {
```

Comment:
Is this thread-safe?

Reply:
I see the assert in `owning_thread_from_monitor` but not `owning_thread_from_object`.